### ISSUE TITLE
cmake: Add option not to define PROJ_LIB.

### ISSUE THREAD

**Issue by c0nk** [OSGeo/PROJ#797]
I am building with CMake and I do not want to have PROJ_LIB defined and that path included in the executable code.

**Comment by kbevers** [comment]
I don't understand this. Please elaborate.

**Comment by c0nk** [comment]
The cmake build always defines PROJ_LIB when building libproj:
```
add_definitions(-DPROJ_LIB="${CMAKE_INSTALL_PREFIX}/${DATADIR}")
```
It would be nice to be able to disable this being set. In my case, there is no data directory on the target system.

**Comment by kbevers** [comment]
Okay, now I understand. I don't think that the PROJ_LIB is compiled into the executables. If I am not mistaken it is only used during install and testing. I could be wrong though.

I am not going to change this myself, but if you or someone else want to submit a pull request with the changes I am happy to look at it. It will not be included in 5.0.0 since we are already too far into the release process to add more changes and features.
